src/lib/Form/Type/ChoiceList/Provider/PermissionAwareContentTypeChoiceListProvider.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Form\Type\ChoiceList\Provider;
+
+use eZ\Publish\API\Repository\PermissionResolver;
+use eZ\Publish\API\Repository\Values\User\Limitation\ContentTypeLimitation;
+use EzSystems\EzPlatformAdminUi\Util\PermissionUtilInterface;
+use eZ\Publish\API\Repository\Values\ContentType\ContentType;
+
+class PermissionAwareContentTypeChoiceListProvider implements ChoiceListProviderInterface
```

Comment:
No need to provide your own interface: `Symfony\Component\Form\ChoiceList\Loader\ChoiceLoaderInterface`